src/processes.c

```diff
@@ -448,60 +431,60 @@ static void ps_list_add (const char *name, const char *cmdline, procstat_entry_t
 		}
 
 		pse->age = 0;
-		pse->num_proc   = entry->num_proc;
-		pse->num_lwp    = entry->num_lwp;
-		pse->vmem_size  = entry->vmem_size;
-		pse->vmem_rss   = entry->vmem_rss;
-		pse->vmem_data  = entry->vmem_data;
-		pse->vmem_code  = entry->vmem_code;
-		pse->stack_size = entry->stack_size;
-		pse->io_rchar   = entry->io_rchar;
-		pse->io_wchar   = entry->io_wchar;
-		pse->io_syscr   = entry->io_syscr;
-		pse->io_syscw   = entry->io_syscw;
-		pse->io_diskr   = entry->io_diskr;
-		pse->io_diskw   = entry->io_diskw;
-		pse->cswitch_vol   = entry->cswitch_vol;
-		pse->cswitch_invol = entry->cswitch_invol;
-
-		ps->num_proc   += pse->num_proc;
-		ps->num_lwp    += pse->num_lwp;
-		ps->vmem_size  += pse->vmem_size;
-		ps->vmem_rss   += pse->vmem_rss;
-		ps->vmem_data  += pse->vmem_data;
-		ps->vmem_code  += pse->vmem_code;
-		ps->stack_size += pse->stack_size;
-
-		ps->io_rchar   += ((pse->io_rchar == -1)?0:pse->io_rchar);
-		ps->io_wchar   += ((pse->io_wchar == -1)?0:pse->io_wchar);
-		ps->io_syscr   += ((pse->io_syscr == -1)?0:pse->io_syscr);
-		ps->io_syscw   += ((pse->io_syscw == -1)?0:pse->io_syscw);
-		ps->io_diskr   += ((pse->io_diskr == -1)?0:pse->io_diskr);
-		ps->io_diskw   += ((pse->io_diskw == -1)?0:pse->io_diskw);
-
-		ps->cswitch_vol   += ((pse->cswitch_vol == -1)?0:pse->cswitch_vol);
-		ps->cswitch_invol += ((pse->cswitch_invol == -1)?0:pse->cswitch_invol);
-		want_init = (entry->vmem_minflt_counter == 0)
-				&& (entry->vmem_majflt_counter == 0);
+		pse->procstat_data.num_proc   = entry->procstat_data.num_proc;
```

Comment:
I wonder in how many cases we can actually pass in a `procstat_data_t *` instead of a pointer to the outer struct... In fact, we may want to factor out the initialization of the inner struct into a separate function.